packages/testing/src/api/index.ts

```diff
@@ -64,6 +67,34 @@ export function initializeTestApp(options: any): admin.app.App {
   );
 }
 
+export function initializeFirestoreTestApp(options: any): FirebaseApp {
+  if (!('projectId' in options)) {
+    throw new Error('projectId not specified');
+  }
+  if (typeof options.auth != 'object') {
+    throw new Error('auth must be an object');
+  }
+  var header = {
+    alg: 'RS256',
+    kid: 'fakekid'
+  };
+  var fakeToken = [
+    base64.encodeString(JSON.stringify(header), /*webSafe=*/ true),
+    base64.encodeString(JSON.stringify(options.auth), /*webSafe=*/ true),
+    'fakesignature'
+  ].join('.');
+  let app = firebase.initializeApp(
+    {
+      projectId: options.projectId,
+      tokenOverride: fakeToken
+    },
+    'app-' + new Date().getTime() + '-' + Math.random()
+  );
+  (app as any).INTERNAL.getToken = () =>
```

Comment:
can you add a comment here explaining that this `INTERNAL.getToken` hijack is to inject the predefined authorization headers into the app (thus bypassing FirebaseAuth)?

Reply:
done